gitapp/lib/HomePage/homepage.dart

```diff
@@ -0,0 +1,119 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:gitapp/constants/constants.dart';
+import 'package:gitapp/user_details/user_details.dart';
+import 'package:gitapp/users_list/User_card.dart';
+
+import 'package:gitapp/users_list/users_list.dart';
+
+class HomePage extends StatefulWidget {
+  HomePage({Key? key, required this.title}) : super(key: key);
+
+  final String title;
+
+  @override
+  HomePageState createState() => HomePageState();
+}
+
+class HomePageState extends State<HomePage> with TickerProviderStateMixin {
+  List<UserCard> _card = <UserCard>[];
+  final UsersList users = new UsersList();
+  String name = "";
+  final TextEditingController _textController = new TextEditingController();
+  bool searching = false;
+
+  Future _getUsers(String text) async {
+    _card.clear();
+
+    if (text != "") {
+      name = text;
+      _textController.clear();
+
+      //users.fillSearchedUsers(text);
+      await users.fetchUsers(text);
+
+      if (users.searchedlist.length != 0) {
+        users.searchedlist.forEach((element) {
+          setState(() {
+            _card.insert(
```

Comment:
There should not be a situation like this.
We can't create a widget for every user which has been found.
Consider that kind of situation. My search result is quite big and I've found 1000 elements. In here you are iterating over this whole collection and you are creating an widget for all of elements. And you are loading 1000 elements into memory. Usage of ListViewBuilder allows us to use much smarter mechanism. ListViewBuilder itself asks for now element every time the users scrolls on it. And in that listviewbuilder we should return a widget only when user will scroll. Thank's to that our app will be much faster. Please refer to documentation, ask Bartek about details or wait for me for tommorow,  we can make a call and I will explain the problem in here.

Reply:
Solved